gnovm/pkg/gnolang/preprocess.go

```diff
@@ -737,6 +737,75 @@ func Preprocess(store Store, ctx BlockNode, n Node) Node {
 				cx := evalConst(store, last, n)
 				return cx, TRANS_CONTINUE
 
+			// TRANS_LEAVE -----------------------
+			case *FuncLitExpr:
+				// We need to closure-call if any variables
+				// used were declared in a for-loop, and this
+				// is the outer-most function within the loop.
+				// NOTE: if the variable is defined as a
+				// function parameter of the closure, then it
+				// is shadowed and not relevant whether or not
+				// the closure is immediately called.
+
+				// Step 1: check if this func lit is directly in
+				// a loop, not embedded within another func lit.
+				// XXX pass store
+				if !isFuncLitDirectlyInLoop(n) {
```

Comment:
this is wrong, it doesn't need to be directly in a loop, it just needs to be in a loop without another func lit intermediary. will fix.